src/js/GlobalSettings/useGlobalSettings.tsx

```diff
@@ -0,0 +1,13 @@
+import * as React from "react";
+import { GlobalSettingsContext } from "./GlobalSettingsProvider";
+
+export default function useGlobalSettings() {
+  const globalSettings = React.useContext(GlobalSettingsContext);
+
+  if (globalSettings === undefined)
+    throw new Error(
```

Comment:
1. the condition can be replaced to `if (!globalSettings)`
2. You added throw of errors to the code, but do you have the place where you are handling it?

Reply:
https://ru.react.js.org/docs/error-boundaries.html

Reply:
Done